tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/Simple.fs

```diff
@@ -59,4 +59,28 @@ module Simple =
         |> asExe
         |> withLangVersionPreview
         |> compileExeAndRun
-        |> shouldSucceed
\ No newline at end of file
+        |> shouldSucceed
+
+
+    [<Theory>]
+    [<InlineData("DateTime", "DateTime.Now")>]
+    [<InlineData("int", "1")>]
+    [<InlineData("Guid", "(Guid.NewGuid())")>]
+    [<InlineData("Byte", "0x1")>]
```

Comment:
Byte-typed literals are 1uy. Curious this compiles and passes. I guess the object is an integer and the first type test fails but the second still succeeds.

Reply:
We should test this kind of logic when testing array types. Particularly signed/unsigned  integer arrays which have strange rules in .net (int32[] can runtime-coerce to unit32[]) . Likewise arrays of enums. Also covariant coercions for reference array types.

Reply:
> Byte-typed literals are 1uy. Curious this compiles and passes. I guess the object is an integer and the first type test fails but the second still succeeds. 

Sigh, yeah, I fixed it locally but didn't push. Well, `obj` is bad :D